cranelift/codegen/src/ir/memflags.rs

```diff
@@ -91,35 +102,101 @@ impl MemFlags {
     }
 
     /// Read a flag bit.
-    const fn read(self, bit: FlagBit) -> bool {
-        self.bits & (1 << bit as usize) != 0
+    const fn read_bit(self, bit: u16) -> bool {
+        self.bits & bit != 0
     }
 
     /// Return a new `MemFlags` with this flag bit set.
-    const fn with(mut self, bit: FlagBit) -> Self {
-        self.bits |= 1 << bit as usize;
+    const fn with_bit(mut self, bit: u16) -> Self {
+        self.bits |= bit;
+        self
+    }
+
+    /// Reads the alias region that this memory operation works with.
+    pub const fn alias_region(self) -> Option<AliasRegion> {
+        // NB: keep in sync with `with_alias_region`
+        match (self.bits & MASK_ALIAS_REGION) >> ALIAS_REGION_OFFSET {
+            0b00 => None,
+            0b01 => Some(AliasRegion::Heap),
+            0b10 => Some(AliasRegion::Table),
+            0b11 => Some(AliasRegion::Vmctx),
+            _ => unreachable!(),
+        }
+    }
+
+    /// Sets the alias region that this works on to the specified `region`.
+    pub const fn with_alias_region(mut self, region: Option<AliasRegion>) -> Self {
+        // NB: keep in sync with `alias_region`
+        let bits = match region {
+            None => 0b00,
+            Some(AliasRegion::Heap) => 0b01,
+            Some(AliasRegion::Table) => 0b10,
+            Some(AliasRegion::Vmctx) => 0b11,
+        };
+        self.bits &= !MASK_ALIAS_REGION;
+        self.bits |= bits << ALIAS_REGION_OFFSET;
         self
     }
 
+    /// Sets the alias region that this works on to the specified `region`.
+    pub fn set_alias_region(&mut self, region: Option<AliasRegion>) {
+        *self = self.with_alias_region(region);
+    }
+
     /// Set a flag bit by name.
     ///
-    /// Returns true if the flag was found and set, false for an unknown flag name.
-    /// Will also return false when trying to set inconsistent endianness flags.
-    pub fn set_by_name(&mut self, name: &str) -> bool {
-        match NAMES.iter().position(|&s| s == name) {
-            Some(bit) => {
-                let bits = self.bits | 1 << bit;
-                if (bits & (1 << FlagBit::LittleEndian as usize)) != 0
-                    && (bits & (1 << FlagBit::BigEndian as usize)) != 0
-                {
-                    false
-                } else {
-                    self.bits = bits;
-                    true
+    /// Returns true if the flag was found and set, false for an unknown flag
+    /// name.  Will also return false when trying to set inconsistent endianness
+    /// flags.
```

Comment:
I think this comment no longer applies; the "Errors" section below covers semantic errors, `set_by_name` doesn't return `false` for them any more.

```suggestion
    /// name.
```